v1/plugin/stream_collector_proxy.go

```diff
@@ -0,0 +1,126 @@
+package plugin
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin/rpc"
+)
+
+type StreamProxy struct {
+	pluginProxy
+
+	plugin StreamCollector
+}
+
+func (c *StreamProxy) StreamMetrics(stream rpc.StreamCollector_StreamMetricsServer) error {
+	// Error channel where we will forward plugin errors to snap where it
+	// can report/handle them.
+	errChan := make(chan string)
+	// Metrics into the plugin from snap.
+	inChan := make(chan []Metric)
+	// Metrics out of the plugin into snap.
+	outChan := make(chan []Metric)
+	err := c.plugin.StreamMetrics(inChan, outChan, errChan)
+	if err != nil {
+		return err
+	}
+	go metricSend(c.plugin, outChan, stream)
+	go errorSend(c.plugin, errChan, stream)
+	streamRecv(c.plugin, inChan, stream)
+	return nil
+}
+
+func errorSend(
+	plugin StreamCollector,
+	ch chan string,
+	s rpc.StreamCollector_StreamMetricsServer) {
+	for r := range ch {
+		reply := &rpc.CollectReply{
+			Error: &rpc.ErrReply{Error: r},
+		}
+		if err := s.Send(reply); err != nil {
+			fmt.Println(err.Error())
```

Comment:
I think that is a good point but outside the scope of this PR. The repo itself has a todo somewhere about converting to a consistent method of logging.

Snap does capture the stdout/error from the plugin and prints it in it's own logs with relevant data so I think this might be ok for now. Do you have an opinion on that?
For reference the output in Snap's logs look like:

```
DEBU[2017-02-22T16:51:26-08:00] Heartbeat received at: 2017-02-22 16:51:26.111847854 -0800 PST  _module=plugin-exec io=stdout plugin=test-rand-streamer
```

There is definitely room for improvement around this. I create #62 to track this and would welcome your comments (either here or there).